diffsmhm/diff_stats/cuda/wprp.py

```diff
@@ -184,17 +186,26 @@ def wprp_serial_cuda(
     wprp_grad : array-like, shape (n_grads, n_rpbins)
         The gradients of the projected correlation function.
     """
-    assert not np.allclose(rpbins_squared[0], 0)
-    _rpbins_squared = np.concatenate([[0], rpbins_squared], axis=0)
+    # check if cupy is available
+    xp = cp.get_array_module(x1)
+    can_cupy = xp is cp
+
+    assert not xp.allclose(rpbins_squared[0], 0)
+    _rpbins_squared = xp.concatenate([xp.array([0]), rpbins_squared], axis=0)
```

Comment:
I guess we are changing the indexing in another PR?

Reply:
I think that's what we had discussed yesterday. You mentioned opening a PR that would have the index changes to wprp cuda kernel

Reply:
Yep. Just confirming. Thanks!